File: Modules/MovieDB/Person/PersonMovieColumns.cs
namespace MovieTutorial.MovieDB.Columns;

[ColumnsScript]
[BasedOnRow(typeof(MovieCastRow))]
public class PersonMovieColumns
{
    [Width(220)]
 public string MovieTitle { get; set; }
    [Width(200)]
    public string Character { get; set; }

}

File: Modules/ServerTypes/MovieDB/PersonMovieColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { MovieCastRow } from "./MovieCastRow";

export interface PersonMovieColumns {
    MovieTitle: Column<MovieCastRow>;
    Character: Column<MovieCastRow>;
}

export class PersonMovieColumns extends ColumnsBase<MovieCastRow> {
    static readonly columnsKey = 'MovieTutorial.MovieDB.Columns.PersonMovieColumns';
    static readonly Fields = fieldsProxy<PersonMovieColumns>();
}